### PR TITLE
feat: now OSCR appears on repository page contributor grid view

### LINKED ISSUE #3883: Feature: Add OSCR to repository page contributor grid view
As a developer, I want to show the OSCR for each contributor in the grid view of the repository contributor page so that I can evaluate their open source activity.

image

### PR DESCRIPTION
**Pull request by nickytonline**
## Description

Now the OSCR appears on the repository page contributor grid view.

## Related Tickets & Documents

Closes #3883
- [x] ~Blocked by https://github.com/open-sauced/api/issues/1018~

## Mobile & Desktop Screenshots/Recordings

**Before**

CleanShot 2024-08-07 at 16 56 20

**After**

CleanShot 2024-08-07 at 22 13 27

### Logged Out

1. Go to any repository contributor page, e.g. http://localhost:3000/s/denoland/deno/contributors?range=30
2. Notice the grid view (default view) displays OSCRs with the lock button for contributors.
3. Log in.
4. Notice the grid view (default view) displays OSCRs for contributors.

### Logged In

1. Go to any repository contributor page, e.g. http://localhost:3000/s/denoland/deno/contributors?range=30
2. Notice the grid view (default view) displays OSCRs for contributors.

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [x] Tier 3
- [ ] Tier 4

### CODE REVIEW

File: lib/hooks/api/useRepositoryContributors.ts [hunk 1]
```diff
@@ -1,12 +1,12 @@
 import useSWR, { Fetcher } from "swr";
 import { publicApiFetcher } from "lib/utils/public-api-fetcher";
+import { OrderDirection } from "lib/utils/sorting";
 
 interface PaginatedResponse {
   readonly data: DbRepoContributor[];
   readonly meta: Meta;
 }
 
-type OrderDirection = "ASC" | "DESC";
```
Comment (nickytonline) [inline lib/hooks/api/useRepositoryContributors.ts:9]: `type OrderDirection = "ASC" | "DESC";` was littered all over the codebase, so I consolidated it's usage everywhere into an exported type.

**Review by jpmcb — APPROVED** [review]
Looks awesome: way to turn this around so quickly. Let me know if you need anything on the API!

**Comment by nickytonline** [comment]
> Looks awesome: way to turn this around so quickly. Let me know if you need anything on the API!

Thanks John! I’ll make the small changes in the API. It’s just returning an `updated_at` for the endpoint getting repo contributor data.

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**